torch/fx/experimental/graph_gradual_typechecker.py

```diff
@@ -184,18 +225,25 @@ def type_check_node(self, n: Node):
         - Transpose
         - Add
         """
+        if n.type is None:
+            n.type = Dyn
+
         if n.op == 'placeholder':
-            if n.type is None:
-                n.type = Dyn
-            else:
-                return n.type
+            return n.type
 
         if n.op == 'call_function':
             if n.target in _INFERENCE_RULES:
                 return _INFERENCE_RULES[n.target](n)
             else:
                 raise RuntimeError(f'No inference rule registered for target {n.target}!')
 
+        if n.op == 'call_module':
+            op_type = getattr(self.traced, str(n.target))
+            if type(op_type) in _INFERENCE_RULES:
+                return _INFERENCE_RULES[type(op_type)](n, op_type)
```

Comment:
We should document somewhere what the schema for inference rules should be since now it's not just the `Node`. For example, without reading this code, I don't know what the `op_type` argument is supposed to be. It seems like it's actually the module instance rather than the type?

Reply:
Yeah it's the module in this case. I'll rename it.